### DSO109 - Programming Foundations - Python
#### Lesson 8 - Python for Data Science

______________

#### Page 11 - Hands-On

You are working for an ecology company, and they have been tracking bison throughout North America. They've collected [data](https://repo.exeterlms.com/documents/V2/DataScience/Data-Wrang-Visual/BisonTracking.zip) on the location, number, genus, and species of bison. They'd like to know some basic information about the bison, to determine whether the species is still in danger or whether it is recovering.

**Please perform the following tasks:**

* **Read in your data as a CSV file** 

In [6]:
import pandas as pd

In [8]:
bison = pd.read_csv('/Users/hannah/Library/CloudStorage/GoogleDrive-gracesnouveaux@gmail.com/My Drive/Bethel Tech/Data Science/DSO109 Programming Foundations – Python/Lesson 8: Python for Data Science/BisonTracking.csv')

*`Pandas` package imported, and its `read_csv()` function used to import provided dataset*

______________

* **Look at the first seven rows of your data**

In [9]:
bison.head(7)

Collection/Population Locality SpecimenNumber  Genus   Species  Length
0                  12 Mile Creek         112916  Bison  antiquus       8
1                  12 Mile Creek         112917  Bison  antiquus       7
2                  12 Mile Creek         112918  Bison  antiquus       7
3                  12 Mile Creek         112919  Bison  antiquus       8
4                  12 Mile Creek         112921  Bison  antiquus       9
5                  12 Mile Creek         112925  Bison  antiquus       9
6        Agate Basin-Agate Basin            458  Bison  antiquus       7

*`head` function from `Pandas` package used to preview first 7 rows of dataset*

______________

* **Look at the last ten rows of your data**

In [10]:
bison.tail(10)

Collection/Population Locality        SpecimenNumber  Genus   Species  \
1097                       Simonsen  SEC. 8, 30", E2, #bb  Bison  antiquus   
1098                       Simonsen               SW15 #3  Bison  antiquus   
1099                       Simonsen               SW15 #3  Bison  antiquus   
1100                       Simonsen               SW37 #9  Bison  antiquus   
1101                       Simonsen               SW37 #9  Bison  antiquus   
1102                         Terapa              TERA 403  Bison  antiquus   
1103                         Terapa              TERA 404  Bison  antiquus   
1104                         Terapa              TERA 426  Bison  antiquus   
1105                           WICA            422F7A5179  Bison     bison   
1106                           WICA      AVID*074*597*636  Bison     bison   

      Length  
1097       8  
1098       8  
1099       7  
1100       9  
1101       8  
1102       7  
1103       8  
1104       9  
1105       7  
1106       7

*`tail` function from `Pandas` package used to preview last 10 rows of dataset*

______________

* **Determine the number of rows and columns your dataset has**

In [16]:
print('There are', len(bison), 'rows in the bison dataset')

There are 1107 rows in the bison dataset


In [17]:
print('There are', len(bison.columns), 'columns in the bison dataset')

There are 5 columns in the bison dataset


*`len` function from `Pandas` package used to confirm how many rows are in the dataset, then used*
*again, in combination with the `.column` `Pandas` function to confirm how many columns there are*

______________

**And answer the following questions:**
* **How many bison are of the species antiquus?**

In [18]:
bison.Species.value_counts()

antiquus        633
bison           435
occidentalis     39
Name: Species, dtype: int64

There are 633 bison from the antiquus species

*`value_counts` function from `Pandas` package used to view frequencies for the `Species` column*

______________

* **What is the mean and standard deviation of Length?**

In [19]:
bison.Length.describe()

count    1107.000000
mean        7.981030
std         0.808299
min         7.000000
25%         7.000000
50%         8.000000
75%         9.000000
max         9.000000
Name: Length, dtype: float64

The mean of Length is: 7.981030
The standard deviation of Length is: 0.808299

*`describe` function from `Pandas` package used to view descriptive statistics for the `Length` column*

______________

* **What is the median length of the bison?**

In [20]:
bison.Length.median()

8.0

The median Length of the bison is: 8

*`median` function from `Pandas` package used to find median for the `Length` column*